### PR TITLE
Ref: Dropdown Menu Revamped

### LINKED ISSUE #542: Improve Portal Dropdown UX
## Description
As a user,
I need a more intuitive UX for the dropdown in Portal,
so that I can see the needed options on hovering.

## Mocks

Screenshot from 2020-05-18 22-39-37
Screenshot from 2020-05-18 22-39-14

### Update [Required]
- [ ] Dropdown should be shown on hovering like the main AnitaB website
- [ ] Some irrelevant options line meetup locations should be removed

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
2-3 days

### PR DESCRIPTION
**Pull request by ritwickraj78**
### Description
- Dropdowns no more need clicks to open.
- List of communities removed from dropdown
- Change in color just like the Anitab.org website on hovering

Fixes #542 

### Type of Change:
**Delete irrelevant options.**

- User Interface

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?

Tested Locally
ezgif com-video-to-gif (2)

### Checklist:
**Delete irrelevant options.**

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by SanketDG — CHANGES_REQUESTED** [review]
Could you squash the first two commits? They seem to represent one feature but are disconnected

File: requirements/dev.txt [hunk 1]
```diff
@@ -14,3 +14,4 @@ djangorestframework==3.8.2
 geopy==1.15.0
 pinax-notifications==5.0.3
 geoip2==3.0.0
+django-ipware==3
```
Comment (SanketDG) [inline requirements/dev.txt]: please pin a specific version

File: .travis.yml [hunk 2]
```diff
@@ -22,6 +22,8 @@ install:
 before_script:
   - export GOOGLE_MAPS_API_KEY=$GOOGLE_MAPS_API_KEY
   - export SECRET_KEY=foobarbaz
+#  - curl -o ipdb.tar.gz https://download.maxmind.com/app/geoip_download?edition_id=GeoLite2-City&license_key=$MAXMIND_LICENSE_KEY&suffix=tar.gz
+#  - tar -xzf ipdb.tar.gz -C systers_portal/
```
Comment (SanketDG) [inline .travis.yml:26]: I am hoping this will be uncommented in the future.

File: systers_portal/GeoLite2-City_20200616/COPYRIGHT.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+Database and Contents Copyright (c) 2020 MaxMind, Inc.
```
Comment (SanketDG) [inline systers_portal/GeoLite2-City_20200616/COPYRIGHT.txt:1]: When we move to doing this to the CI we don't need this right?

File: systers_portal/meetup/tests/test_views.py [hunk 4]
```diff
@@ -1129,3 +1130,113 @@ def test_post_delete_support_request_comment_view(self):
...
+        self.meetup = Meetup.objects.create(title='Foo Bar Baz', slug='foo-bar-baz',
+                                            date=(timezone.now() + timezone.timedelta(4)).date(),
+                                            time=timezone.now().time(),
+                                            description='This is test Meetup',
+                                            venue='Foo Systers',
+                                            meetup_location=self.location,
+                                            created_by=self.systers_user,
+                                            leader=self.systers_user,
+                                            last_updated=timezone.now())
+        self.meetup2 = Meetup.objects.create(title='Foo Baz', slug='foobar',
```
Comment (SanketDG) [inline systers_portal/meetup/tests/test_views.py:1155]: nit: spacing between each of the objects

File: systers_portal/meetup/tests/test_views.py [hunk 5]
```diff
@@ -1129,3 +1130,113 @@ def test_post_delete_support_request_comment_view(self):
...
+                                             meetup_location=self.location,
+                                             created_by=self.systers_user,
+                                             leader=self.systers_user,
+                                             last_updated=timezone.now())
+        self.meetup3 = Meetup.objects.create(title='Foob Baz', slug='foobarbaz',
+                                             date=(timezone.now() + timezone.timedelta(2)).date(),
+                                             time=timezone.now().time(),
+                                             description='This is test Meetup',
+                                             venue='Foo Systers',
+                                             meetup_location=self.location1,
```
Comment (SanketDG) [inline systers_portal/meetup/tests/test_views.py]: why do you need `location1` why not just `location`?

File: systers_portal/meetup/urls.py [hunk 6]
```diff
@@ -32,8 +33,8 @@
         ApproveRequestMeetupView.as_view(), name="approve_meetup_request"),
     url(r'^(?P<meetup_slug>[\w-]+)/reject_meetup_request/$',
         RejectMeetupRequestView.as_view(), name="reject_meetup_request"),
-    # url(r'^all-upcoming-meetups/search/$', UpcomingMeetupsSearchView.as_view(),
-    #     name='search_meetups'),
+    url(r'^all-upcoming-meetups/search/$', UpcomingMeetupsSearchView.as_view(),
```
Comment (SanketDG) [inline systers_portal/meetup/urls.py]: I wish we could change this to `/meetups/all/search` :(

File: systers_portal/static/js/search_menu.js [hunk 7]
```diff
@@ -18,45 +18,41 @@ function getCookie(name) {
...
+		    var x = "<h3>No Meetups Found</h3>";
+		  }
+			else{
+			  var x = "<div class=\"row\">";
 			for(var i of a.search_results){
-				x += "<div class=\n" +
-				"'text-bottom ml15'>"+i.location +"<div><a href='../"+i.location_slug+"\n" +
-				"/"+i.meetup_slug+"''>"+i.meetup+"</a>"+"</div>"+i.date+"\n" +
-				"<div><b>"+i.distance+" "+i.unit+"</b></div></div>";			
+				x = x + "<div class=\"col-sm-6 col-md-4\">" +
```
Comment (SanketDG) [inline systers_portal/static/js/search_menu.js]: Can we refactor this to use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals ?

Way more maintainable imo, please open an issue if this is not in the scope.

File: systers_portal/templates/meetup/list_meetup.html [hunk 8]
```diff
@@ -5,35 +5,47 @@
...
-        {% for city in cities_list %}
-        <option value = "{{forloop.counter}}">{{ city }}</option>
-
-        {% endfor %}
-          </select>
-        
-        <div id="location" class="fake-border hidden">
-        <i class="glyphicon glyphicon-map-marker"></i>
-
+<style type="text/css"></style>
```
Comment (SanketDG) [inline systers_portal/templates/meetup/list_meetup.html]: let's remove this?

**Review by SanketDG — CHANGES_REQUESTED** [review]
I will run this locally and we should be good for merge then :100:

File: systers_portal/GeoLite2-City_20200616/COPYRIGHT.txt (hunk 3 above)
Comment (ritwickraj78) [reply systers_portal/GeoLite2-City_20200616/COPYRIGHT.txt:1]: Yup, but when we move to the CI, the extracted folder would contain this.

File: systers_portal/meetup/tests/test_views.py (hunk 4 above)
Comment (ritwickraj78) [reply systers_portal/meetup/tests/test_views.py:1155]: Will Take Care of it

File: systers_portal/meetup/urls.py (hunk 6 above)
Comment (ritwickraj78) [reply systers_portal/meetup/urls.py]: Sure I will do it

File: systers_portal/static/js/search_menu.js (hunk 7 above)
Comment (ritwickraj78) [reply systers_portal/static/js/search_menu.js]: The link you shared, how is different from what is done right now? Didnt get you

File: systers_portal/templates/meetup/list_meetup.html (hunk 8 above)
Comment (ritwickraj78) [reply systers_portal/templates/meetup/list_meetup.html]: Oh Yes We have to remove it

File: systers_portal/meetup/tests/test_views.py (hunk 5 above)
Comment (ritwickraj78) [reply systers_portal/meetup/tests/test_views.py]: I will check it once and let you know

File: .travis.yml (hunk 2 above)
Comment (ritwickraj78) [reply .travis.yml:26]: Yaa maybe, if the download works

File: requirements/dev.txt (hunk 1 above)
Comment (ritwickraj78) [reply requirements/dev.txt]: I will look into it and add one :)

File: systers_portal/templates/base.html [hunk 9]
```diff
@@ -209,6 +204,7 @@
 <script type="text/javascript" src="{% static 'js/community-search.js' %}"></script>
 <script src="{% static 'js/choose_profile_pic.js' %}"></script>
 <script src="{% static 'js/search_menu.js' %}"></script>
+<script src="{% static 'js/main.js' %}"></script>
```
Comment (sammy1997) [inline systers_portal/templates/base.html]: What was the need for main.js? @ritwickraj78

Comment (ritwickraj78) [reply systers_portal/templates/base.html]: Ohk I had removed the file but this was left here itself. Needs to be removed

File: requirements/dev.txt (hunk 1 above)
Comment (ritwickraj78) [reply requirements/dev.txt]: Resolved

File: systers_portal/meetup/tests/test_views.py (hunk 4 above)
Comment (ritwickraj78) [reply systers_portal/meetup/tests/test_views.py:1155]: Done

File: systers_portal/meetup/tests/test_views.py (hunk 5 above)
Comment (ritwickraj78) [reply systers_portal/meetup/tests/test_views.py]: Checked it. It was not needed. Removed it

File: systers_portal/meetup/urls.py (hunk 6 above)
Comment (ritwickraj78) [reply systers_portal/meetup/urls.py]: Done :)

File: systers_portal/templates/meetup/list_meetup.html (hunk 8 above)
Comment (ritwickraj78) [reply systers_portal/templates/meetup/list_meetup.html]: Removed

File: systers_portal/templates/base.html (hunk 9 above)
Comment (ritwickraj78) [reply systers_portal/templates/base.html]: Removed!

File: systers_portal/GeoLite2-City_20200616/COPYRIGHT.txt (hunk 3 above)
Comment (SanketDG) [reply systers_portal/GeoLite2-City_20200616/COPYRIGHT.txt:1]: Right, but in the CI right?

So it wouldn't stay back in the repo right?

File: systers_portal/static/js/search_menu.js (hunk 7 above)
Comment (SanketDG) [reply systers_portal/static/js/search_menu.js]: Can you go through the link again? It's  basically like

```
`hello ${name}`
```
vs
```
"hello " + name
```

File: systers_portal/GeoLite2-City_20200616/COPYRIGHT.txt (hunk 3 above)
Comment (ritwickraj78) [reply systers_portal/GeoLite2-City_20200616/COPYRIGHT.txt:1]: If it is downloaded during the Build, then no it wont be present.

File: systers_portal/static/js/search_menu.js (hunk 7 above)
Comment (ritwickraj78) [reply systers_portal/static/js/search_menu.js]: Ohk kind of formatting.

File: systers_portal/GeoLite2-City_20200616/COPYRIGHT.txt (hunk 3 above)
Comment (SanketDG) [reply systers_portal/GeoLite2-City_20200616/COPYRIGHT.txt:1]: Right, so ideally in a future commit, you would remove these files when the CI is working and also provide a `manage` command to download the database manually for developers.

File: systers_portal/meetup/views.py [hunk 10]
```diff
@@ -801,79 +774,50 @@ def post(self, request):
...
-#                                 'location_slug': meetup.meetup_location.slug,
-#                                 'meetup_slug': meetup.slug,
-#                                 'distance': distance,
-#                                 'unit': unit})
-#
-#             results.sort(key=operator.itemgetter('date'))
-#             return JsonResponse({'search_results': results}, safe=False)
+
+class UpcomingMeetupsSearchView(ListView):
+    """Search Upcoming Meetups By  Keyword and Filter Date and Distance"""
```
Comment (sakshi1499) [inline systers_portal/meetup/views.py:779]: Minor Change: Correct the spacing.

File: systers_portal/templates/meetup/list_meetup.html [hunk 11]
```diff
@@ -5,35 +5,46 @@
...
-        <option value = "{{forloop.counter}}">{{ city }}</option>
-
-        {% endfor %}
-          </select>
-        
-        <div id="location" class="fake-border hidden">
-        <i class="glyphicon glyphicon-map-marker"></i>
-
+<h1 class="mt40 text-center meetup-list-head">Upcoming Meetups</h1>
+<p class="meetup-list-subhead">We host meetups, workshops, networking events, and more. See what’s happening near you.</p>
```
Comment (sakshi1499) [inline systers_portal/templates/meetup/list_meetup.html:9]: Honestly you should let the spaces be and not let such changes be reflected.

File: .travis.yml (hunk 2 above)
Comment (sakshi1499) [reply .travis.yml:26]: +1

**Review by sakshi1499 — COMMENTED** [review]
All in all no major changes from my side.

File: systers_portal/static/css/style.css [hunk 12]
```diff
@@ -771,3 +771,11 @@ input[id=go-btn]{
...
 font-family: 'Cairo';
 }
+li[id=nav-menu]:hover{
+background-color:#54bceb;
+}
+@media only screen and (min-width: 768px) {
+  .dropdown:hover .dropdown-menu {
+    display: block;
+  }
+}
```
Comment (SanketDG) [inline systers_portal/static/css/style.css:783]: I rendered the search view on Mobile

[image](https://user-images.githubusercontent.com/8980971/86935716-c4ba9100-c15a-11ea-9214-90a8ac854c7c.png)

Also, when you hover over a menu, the colors change to something unreadable

File: systers_portal/templates/base.html [hunk 13]
```diff
@@ -37,17 +37,12 @@
     <div class="collapse navbar-collapse">
       <ul class="nav navbar-nav text-uppercase" id='lower-nav'>
         {% url 'about-us' as url %}
-        <li {% if url == request.path %}class="active"{% endif %}>
+        <li id="nav-menu" {% if url == request.path %}class="active"{% endif %}>
           <a href="{{ url }}">About Us</a>
         </li>
-        <li class="dropdown {% if '/community/' in request.path or url == request.path %}active{% endif %}" id="community-dropdown">
+        <li id="nav-menu" class="dropdown {% if '/community/' in request.path or url == request.path %}active{% endif %}" id="community-dropdown">
```
Comment (SanketDG) [inline systers_portal/templates/base.html:43]: This is not really needed, and I find the way it is done through Django templates not elegant.

**Review by SanketDG — CHANGES_REQUESTED** [review]
Make sure to squash your commits!

File: systers_portal/static/css/style.css (hunk 12 above)
Comment (sakshi1499) [reply systers_portal/static/css/style.css:783]: Also on this point, I'd like to add the fact that it was supposed to look like [this](https://anitab.org/) and when you hover the black color should actually change to white and not just the change in the background color.

File: systers_portal/meetup/views.py (hunk 10 above)
Comment (ritwickraj78) [reply systers_portal/meetup/views.py:779]: @sakshi1499 That is the part of the formatting IG

File: systers_portal/templates/meetup/list_meetup.html (hunk 11 above)
Comment (ritwickraj78) [reply systers_portal/templates/meetup/list_meetup.html:9]: That complete part is being removed. That is why I removed the space after that as well

File: systers_portal/meetup/views.py (hunk 10 above)
Comment (sakshi1499) [reply systers_portal/meetup/views.py:779]: Between By and Keyword in the Docstring

Comment (ritwickraj78) [reply systers_portal/meetup/views.py:779]: Oh ok nice catch, didnt see that. Thank you :)

File: systers_portal/templates/meetup/list_meetup.html (hunk 11 above)
Comment (sakshi1499) [reply systers_portal/templates/meetup/list_meetup.html:9]: Its okay :)

File: systers_portal/static/css/style.css (hunk 12 above)
Comment (SanketDG) [reply systers_portal/static/css/style.css:783]: @ritwickraj78 Did you look into this?

Comment (sakshi1499) [reply systers_portal/static/css/style.css:783]: Exactly 😬

Comment (ritwickraj78) [reply systers_portal/static/css/style.css:783]: Screenshot from 2020-07-09 12-23-34
@sakshi1499 Could you elaborate a bit more? Didnt get you exactly

Comment (sakshi1499) [reply systers_portal/static/css/style.css:783]: @ritwickraj78 The website that I mentioned is anitaB's official website. Actually, when we hover the blue background on the navbar items has black colored text right? I was saying like the websites of anitaB, make that text white. So add the property of color: white;

File: .travis.yml (hunk 2 above)
Comment (SanketDG) [inline .travis.yml:26]: I have opened https://github.com/anitab-org/portal/issues/584

File: requirements/dev.txt [hunk 14]
```diff
@@ -13,3 +13,5 @@ selenium==3.12.0
 djangorestframework==3.8.2
 geopy==1.15.0
 pinax-notifications==5.0.3
+geoip2==3.0.0
+django-ipware==3.0.0
```
Comment (SanketDG) [inline requirements/dev.txt]: minor: you can remove this

File: systers_portal/meetup/views.py [hunk 15]
```diff
@@ -801,79 +774,50 @@ def post(self, request):
...
+            for meetup in searched_meetups:
+                distance = ''
+                geolocator = Nominatim(timeout=6)
+                g = GeoIP2()
+                if location == "Current Location":
+                    client_ip, is_routable = get_client_ip(request)
+                    if is_routable:
+                        lat, long = g.lat_lon(client_ip)
+                    else:
+                            lat, long = g.lat_lon("google.com")
```
Comment (SanketDG) [inline systers_portal/meetup/views.py]: indentation

File: systers_portal/meetup/views.py [hunk 16]
```diff
@@ -801,79 +774,50 @@ def post(self, request):
...
+                    user_point = Point((float)(long),
+                                       (float)(lat))
+                else:
+                    user_loc = (geolocator.geocode(location))
```
Comment (SanketDG) [inline systers_portal/meetup/views.py]: minor: extra brackets

File: systers_portal/meetup/views.py [hunk 17]
```diff
@@ -801,79 +774,50 @@ def post(self, request):
...
+                    user_point = Point((float)(user_loc.raw['lon']), (float)(user_loc.raw['lat']))
```
Comment (SanketDG) [inline systers_portal/meetup/views.py]: minor: extra brackets.

File: systers_portal/meetup/views.py [hunk 18]
```diff
@@ -801,79 +774,50 @@ def post(self, request):
...
+                meetup_loc = (geolocator.geocode(meetup.meetup_location))
+                meetup_point = Point((float)(meetup_loc.raw['lon']),
+                                     (float)(meetup_loc.raw['lat']))
+                distance = (int)(user_point.distance(meetup_point)) * 100
```
Comment (SanketDG) [inline systers_portal/meetup/views.py]: extra brackets, please note that these are just functions, what you are doing is basically evaluating an expression `(int)` and then calling on it

File: systers_portal/meetup/views.py [hunk 19]
```diff
@@ -801,79 +774,50 @@ def post(self, request):
...
+                results.sort(key=operator.itemgetter('distance'))
```
Comment (SanketDG) [inline systers_portal/meetup/views.py]: what's the point of doing  this?

File: systers_portal/meetup/views.py [hunk 20]
```diff
@@ -801,79 +774,50 @@ def post(self, request):
...
+                unit = 'kilometers from your location'
+
+                results.append({'date': meetup.date,
+                                'meetup': meetup.title,
+                                'distance': distance,
+                                'location': meetup.meetup_location.name,
+                                'meetup_slug': meetup.slug,
+                                'unit': unit})
```
Comment (SanketDG) [inline systers_portal/meetup/views.py]: If you are defining unit once, then you can just pass it once in the response json

You can also implement these lines as a dict comprehension if you remove unit

**Review by SanketDG — CHANGES_REQUESTED** [review]
Some more comments.

As for the location data can you move it to a dedicated folder like `locationdata/` instead of keeping at the top level?

File: systers_portal/meetup/views.py (hunk 19 above)
Comment (ritwickraj78) [reply systers_portal/meetup/views.py]: Forgot to remove it.

File: systers_portal/meetup/views.py (hunk 20 above)
Comment (ritwickraj78) [reply systers_portal/meetup/views.py]: Will be done:)

File: systers_portal/meetup/views.py (hunk 18 above)
Comment (ritwickraj78) [reply systers_portal/meetup/views.py]: Sure, will take care of it

File: requirements/dev.txt (hunk 14 above)
Comment (ritwickraj78) [reply requirements/dev.txt]: Removing it would cause the code to break. What do you suggest?

File: .travis.yml (hunk 2 above)
Comment (ritwickraj78) [reply .travis.yml:26]: Great

File: systers_portal/meetup/views.py (hunk 15 above)
Comment (ritwickraj78) [reply systers_portal/meetup/views.py]: Done

File: requirements/dev.txt (hunk 14 above)
Comment (SanketDG) [reply requirements/dev.txt]: But this file inherit's from `prod.txt` right, apologies if I am not seeing something.

Comment (ritwickraj78) [reply requirements/dev.txt]: Oh I did not see the first line of dev.txt till date. Sorry

**Review by sammy1997 — COMMENTED** [review]
LGTM. @SanketDG @sakshi1499 ?

**Review by sakshi1499 — APPROVED** [review]

**Review by SanketDG — APPROVED** [review]

**Review by sammy1997 — APPROVED** [review]
LGTM

**Pull request merged**